BrightFutures/ExecutionContext.swift

```diff
@@ -50,9 +50,22 @@ public func toContext(queue: dispatch_queue_t) -> ExecutionContext {
     return Queue(queue: queue).context
 }
 
-typealias ThreadingModel = () -> ExecutionContext
+/// Defines BrightFutures' default threading behaviour, by default this is
+/// set to `StandardThreadingModel.AsynchronousOnCurrentThread`
+public var DefaultThreadingModel: StandardThreadingModel = .AsynchronousOnCurrentThread
```

Comment:
I think we should have here an abstraction. Not ```StandardThreadingModel```, but rather public protocol ```ThreadingModelType``` with ```var context: ExecutionContext {get}``` single member. Like this it will be easily possible to implement the protocol outside and set globally without even knowing of ```StandardThreadingModel```.

```StandardThreadingModel``` just implements the same protocol.

Hope it's clear enough what I mean and makes sense to you.